File: AlgorithmsPackPythonSolutions/376. Wiggle Subsequence.py
class Solution:
    def wiggleMaxLength(self, nums: List[int]) -> int:
        up = down = 0
        for idx in range(1, len(nums)):
            if nums[idx-1] < nums[idx]:
                up = down + 1
            elif nums[idx-1] > nums[idx]:
                down = up + 1
        return 1 + max(up, down)

File: AlgorithmsPackCPP/376. Wiggle Subsequence.cpp
class Solution {
public:
    int wiggleMaxLength(vector<int>& nums) {
        int up = 0, down = 0;
        for(int idx = 1; idx < nums.size(); idx++){
            if(nums[idx-1]<nums[idx])
                up = down+1;
            else if(nums[idx-1]>nums[idx])
                down = up+1;
        }
        return 1+max(up,down);
    }
};